### DISCUSSION TITLE
IP address K3S goes weird

### DISCUSSION THREAD

**Discussion by rthamrin** [k3s-io/k3s#4857] [Show and tell]
I run k3s version v1.21.7+k3s1 (ac705709) on ubuntu 21.10
I set internal and external IP addresses through k3s.service.

```
#vim /etc/systemd/system/k3s.service 
```
everything goes right until I try to remove the IP address. I remove the setting that I set before, and restart everything, but seems the IP address won't go.

```
#restart daemon-reload
#restart k3s.service
#restart NetworkManager
#reboot
even disable the Ethernet 
```

I checked through
```
#kubectl get nodes -o wide
```
the IP address is still there. how can I remove the IP address?

NB: I have some K3S-agent connected to internal IP address

**Comment by brandond** [comment]
Restarting without the --node-external-ip flag won't cause the annotations or address values to be unset. What specifically are you trying to do?

**Reply by rthamrin** [reply]
I want to remove my external-Ip address from my node, my 192.168.77.112 IP address

```
# kubectl get nodes -o wide

NAME            STATUS     ROLES                  AGE   VERSION        INTERNAL-IP     EXTERNAL-IP      OS-IMAGE       KERNEL-VERSION      CONTAINER-RUNTIME
68
master          Ready      control-plane,master   70d   v1.21.7+k3s1   192.168.43.1  192.168.77.112   Ubuntu 21.10   5.13.0-22-generic   containerd://1.4.12-k3s1
# 
```

**Reply by brandond** [reply]
There's not currently a good way to unset that once you've set it. You might be able to get away with deleting the node and then restarting k3s to recreate it, but I would test that somewhere non-critical first.

**Reply by rthamrin** [reply]
is there any update for this?
